### PR TITLE
fix(select): make select with no options return undefined when OK is pressed

### PR DESCRIPTION
**Pull request by zakton5**
#### Short description of what this resolves:
Opening a select without options in it and then pressing Okay resulted in the bound model being set to `{}`

#### Changes proposed in this pull request:
The current code instantiates an empty object and attempts to put values of any text inputs in that object. I've added a check to determine if that object has any properties in it and, if it does not, it returns `undefined` instead of that empty object. 

**Ionic Version**:3.x

**Fixes**: #10435

### CODE REVIEW

File: src/components/alert/alert-component.ts [hunk 1]
```diff
@@ -322,7 +322,9 @@ export class AlertCmp {
     this.d.inputs.forEach(i => {
       values[i.name] = i.value;
     });
-    return values;
+
```
Comment (manucorporat) [inline src/components/alert/alert-component.ts]: could we implement this with an early return instead?

```
if (this.d.inputs.length === 0) {
  return undefined;
}
```
this way we don't allocate `values` and Object.keys()'s array.

Comment (manucorporat) [inline src/components/alert/alert-component.ts]: cc @zakton5

Comment (zakton5) [inline src/components/alert/alert-component.ts]: Yeah that sounds like a better idea. I'll commit that soon.

**Comment by manucorporat** [comment]
Merged!!! thanks a lot for the PR! @zakton5 !!

**Comment by zakton5** [comment]
No problem :) @manucorporat

**Pull request merged**